assignment/Apex/AccountContactService.apxc.txt

```diff
@@ -0,0 +1,46 @@
+public class AccountContactService {
+    
+    public static void addContactIdToAccount(Id contactId){
+        String contactIdLikeClause = '%'+contactId+'%';
+        Account accountDetail;
+        try{
+        	accountDetail = [select Id, ContactIDs__c from Account where ContactIDs__c like :contactIdLikeClause];
+    	}catch(System.QueryException e){
+            accountDetail = null;
+    	}
+        
+        //if contactid is already present with account then return 
+        if(accountDetail != null){
+            return;
+        }
+        
+        Contact contactDetail = fetchContactById(contactId);
+        //if contact is not valid then return
+        if( contactDetail == null || contactDetail.AccountId == null){
+            return;
+        }
+        
+        Account accountDetailToUpdated = [select Id, ContactIDs__c from Account where Id =:contactDetail.AccountId];
+        addContactIdsToAccount(accountDetailToUpdated, contactDetail);
+        update accountDetailToUpdated;
+    }
+    
+    public static Contact fetchContactById(Id contactId){
+        return [select Id, AccountId from Contact where Id =:contactId];
+    }
+    
+    public static void addContactIdsToAccount(Account accountDetailToUpdated, Contact contactDetail){
+        String contactIds = '';
+        //fetching all contact mapped with accountid then appending id in contactIds which will be updaed to account
+        for(Contact cont : [select Id from Contact where AccountId =:contactDetail.AccountId]){
```

Comment:
We need to check if `ContactIDs__c ` is null that we have to add all contact ids in it but If it have some values than we need to add just a contact id it it.